modules/internal/DefaultRectangular.chpl

```diff
@@ -627,11 +621,13 @@ module DefaultRectangular {
     proc dsiHasSingleLocalSubdomain() param return true;
 
     proc dsiLocalSubdomain() {
-      return _newDomain(this);
+      // TODO: this should only return this way if this.locale == here
```

Comment:
I agree it's easy but I didn't want to add it to this PR.  I'll try to open up a new PR or issue for it in order to remove this TODO.